### PR TITLE
New implementation of wasserstein loss based on the absolute difference

### PR DESCRIPTION
**Pull request by NinaWie**
This is a really cool project and nice code base! 

I read the paper and noticed that in this setting, the Wasserstein distance should be equivalent to the absolute difference between the gt number and predicted numbers (weighted by softmax probabilities). This is due to the fact that the ground truth distribution is a one hot vector, so Optimal Transport is not needed to compute the optimal coupling between the bins of the distributions. 

To test this, I implemented this new version of the loss function, and tested if it is indeed equivalent to the Wasserstein distance. The loss is implemented as a new class `AbsDiffNumberTokenLoss`, with the same structure as your loss functions. For the tests, I copied the tests that you had for the Wasserstein loss and modified them in some places to assert that the result of `AbsDiffNumberTokenLoss` is equivalent to the result of `WassersteinNumberTokenLoss`.

The main advantages of this new version is that it should be more efficient and that it doesn't require equally-spaced and sorted number tokens. Let me know if it requires more explanation or if you discover any issues with this implementation!

### CODE REVIEW

**Review by jannisborn — APPROVED** [review]
Wow, this is fantastic 🤩  Just added a unittest for speed comparison and it gives a 500x speedup. On M3 Mac, I get 10ms instead of 6.1 seconds for a realistic batch, that is huge!

```sh
****Speed Average for BS x T x V = 128 x 300 x 32448***
        Wasserstein: 6.1025
        Absdiff: 0.0119
```
